Brooks/LanguageRenderer/NewCSharpRenderer.hs

```diff
@@ -0,0 +1,561 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module LanguageRenderer.NewCSharpRenderer (
+    -- * C# Code Configuration -- defines syntax of all C# code
+    CSharpCode(..)
+) where
+
+import New (Label,
+    PackageSym(..), RenderSym(..), KeywordSym(..), PermanenceSym(..),
+    BodySym(..), BlockSym(..), ControlBlockSym(..), StateTypeSym(..),
+    UnaryOpSym(..), BinaryOpSym(..), ValueSym(..), NumericExpression(..), 
+    BooleanExpression(..), ValueExpression(..), Selector(..), FunctionSym(..), 
+    SelectorFunction(..), StatementSym(..), ControlStatementSym(..), ScopeSym(..),
+    MethodTypeSym(..), ParameterSym(..), MethodSym(..), StateVarSym(..), 
+    ClassSym(..), ModuleSym(..))
+import NewLanguageRenderer (Terminator(..), fileDoc', 
```

Comment:
All those trailing `DocD` are for weird historical reasons, and should probably all be removed.

Reply:
(I mean from the name)

Reply:
If I just drop `DocD`, we will end up with a lot of function names that are the same as function names in the symantics typeclasses. For example, `packageDocD` would become `package` which is the same as the `PackageSym` function for creating a package. In these cases, can I just drop the `D` and have the name as `packageDoc`? But then for consistency I would want to keep `Doc` in the names for all of these functions. I think it makes sense to have `Doc` in these names, since they are all functions for creating `Doc`s.